src/gc.c

```diff
@@ -3776,7 +3793,8 @@ static void *gc_managed_realloc_(jl_ptls_t ptls, void *d, size_t sz, size_t olds
     SetLastError(last_error);
 #endif
     errno = last_errno;
-    maybe_record_alloc_to_profile((jl_value_t*)b, sz, jl_gc_unknown_type_tag);
+    // gc_managed_realloc_ is currently used exclusively for resizing array buffers.
+    maybe_record_alloc_to_profile((jl_value_t*)b, sz, (jl_datatype_t*)jl_buff_tag);
```

Comment:
Since this is realloc should we record the delta?

Reply:
ooh yeah great idea! 💡

Reply:
```suggestion
    maybe_record_alloc_to_profile((jl_value_t*)b, sz - oldsz, (jl_datatype_t*)jl_buff_tag);
```

Do you think we should use the supplied `sz` or the `allocsz`, computed after alignment? For pool allocated objects, we're using the `osize` not the object's actual size. So `allocsz - oldsz` seems most similar here to me?

Reply:
Do you think we should use the supplied `sz` or the `allocsz`, computed after alignment? For pool allocated objects, we're using the `osize` not the object's actual size. So `allocsz - oldsz` seems most similar here to me?

```suggestion
    maybe_record_alloc_to_profile((jl_value_t*)b, allocsz - oldsz, (jl_datatype_t*)jl_buff_tag);
```

Reply:
@gbaraldi's suggestion:
- Let's report the delta,
- and only record the alloc if the delta is positive.

Reply:
Going with this as the final change:
```suggestion
    if (allocsz > oldsz)
        maybe_record_alloc_to_profile((jl_value_t*)b, allocsz - oldsz, (jl_datatype_t*)jl_buff_tag);
    end
```